### PR TITLE
RD2: Shiny and Amazing Custom User Interface for RD Entry

### PR DESCRIPTION
**Pull request by force2b**
# Critical Changes
- For orgs that have migrated to Enhanced Recurring Donations, to use the new custom User Interface it will be necessary to enable New and Edit overrides on the Recurring Donations object to point to the `Rd2_EntryForm` component as well as to grant permissions to the `RD2_EntryFormController` apex class. See the documentation for more information and steps.

# Changes
- A new custom user interface has been added for Enhanced Recurring Donations.

# Issues Closed

# Community Ideas Delivered

# New Metadata

Custom Label
- lblCurrency
- RD2_EntryFormHeader
- RD2_EntryFormCustomFieldsSectionHeader
- RD2_EntryFormDonationSectionHeader
- RD2_EntryFormDonorSectionHeader
- RD2_EntryFormOtherSectionHeader
- RD2_EntryFormDonorTypeHelpText
- RD2_EntryFormDonorTypeLabel
- RD2_EntryFormInsertSuccessMessage
- RD2_EntryFormUpdateSuccessMessage
- RD2_EntryFormPeriodAdvanced
- RD2_EntryFormPeriodHelpText
- RD2_EntryFormPeriodLabel
- RD2_EntryFormPeriodPluralDaily
- RD2_EntryFormPeriodPluralMonthly
- RD2_EntryFormPeriodPluralWeekly
- RD2_EntryFormPeriodPluralYearly
- RD2_EntryFormScheduleEveryLabel
- RD2_EntryFormMissingPermissions

Aura Component
- RD2_EntryForm

LWC
- rd2EntryForm
- rd2EntryFormScheduleSection
- rd2EntryFormDonorSection
- rd2EntryFormCustomFieldsSection

Apex Class
- RD2_EntryFormController
- RD2_EntryFormController_TEST

# Deleted Metadata

# Unaggregated Pull Requests
* RD2:  Create schedule section LWC [[PR5521](https://github.com/SalesforceFoundation/NPSP/pull/5521)]
* RD2 Custom Form: Donor Type Selection [[PR5557](https://github.com/SalesforceFoundation/NPSP/pull/5557)]
* RD2: Custom Field Section UI [[PR5609](https://github.com/SalesforceFoundation/NPSP/pull/5609)]
* RD2 UI: Advanced Scheduling [[PR5599](https://github.com/SalesforceFoundation/NPSP/pull/5599)]

### CODE REVIEW

File: src/labels/CustomLabels.labels [hunk 1]
```diff
@@ -7947,6 +7998,14 @@ If preferred, you can disable Gift Entry and instead use older Batch Gift Entry
         <protected>false</protected>
         <shortDescription>lblCountry</shortDescription>
         <value>Country</value>
+    </labels>
+     <labels>
+        <fullName>lblCurrency</fullName>
```
Comment (force2b) [inline src/labels/CustomLabels.labels:8003]: @andrewyu-salesforce I don't think this is necessary, plus it creates an alternative translation for a standard field. It should be possible to retrieve the standard translation for the CurrencyIsoCode field dynamically using getLabel(). Can you create a PR to fix this? It's something that we'd need to do before the new label is merged into master. Once that happens there's no way to remove it from our customers orgs.

Comment (RadSFDeveloper) [reply src/labels/CustomLabels.labels:8003]: The reason for a new label was the that getLabel is displaying "Currency ISO Code" while the standard Salesforce layout displays the "Currency" label. However, it is perfectly fine to use the field label.

Comment (force2b) [reply src/labels/CustomLabels.labels:8003]: Thanks @RadSFDeveloper. I'll pull up a multi-currency org to review the labels just to make sure that we get the label we want if using the standard field label.

Comment (force2b) [reply src/labels/CustomLabels.labels:8019]: So, as much as I don't like this you're 100% right - for now we need to keep this custom label.

Calling `System.debug(Opportunity.CurrencyIsoCode.getDescribe().getLabel())` returns the following each language:
- English: Currency ISO Code
- Spanish: Código ISO de divisa
- German: ISO-Währungskürzel
- French: Code ISO de devise
- Japanese: 通貨 ISO コード

Yet, every page layout with the Currency Code on it always shows a field label of "Currency" regardless of the current users language and clearly not matching the result of getLabel() against the CurrencyIsoCode describe.

Our choices are to either display it as "Currency ISO Code" (and the translations) or user our own label so it more closely matches what Users would actually see on a page layout.

@andrewyu-salesforce  Let's just keep the label. Sorry for the back-and-forth on it. I'm going to reach out the core localization team to see what I can find out.

**Comment by sam-knox** [comment]
@force2b - Will only existing customers need to take the enablement steps here, or will new customers need to as well? In other words, are we planning to add the button and class assignment steps to MetaDeploy?

**Comment by force2b** [comment]
@sam-knox  MetaDeploy does handle the New/Edit button overrides, and it _will_ (though it doesn't at the moment because we forgot about it) handle granting permissions for the controller apex class to the System Administrator and Standard User profiles only. Admins will still be required to grant permissions to this class for any custom profiles in use.

Any org that previously enabled RD2 prior to 3.180 will need to do the above steps manually in their org.

**Comment by sam-knox** [comment]
@force2b - Perfect, thanks!

**Pull request merged**